Add endpoint documentation for recoverCumulusMessages (#319)
nasa/cumulus-api: content/dead-letter-archive.md

## Recover cumulus messages

> Endpoint provides a mechanism for recovery of S3 sfEventSqsToDbRecords dead letter objects (created as described in the [Core Documentation](https://nasa.github.io/cumulus/docs/features/dead_letter_archive)).   The endpoint will invoke an async operation that will attempt to process all of the records in the specified location.

Reviewer: ```suggestion
Endpoint provides a mechanism for recovery of S3 sfEventSqsToDbRecords dead letter objects (created as described in the [Core Documentation](https://nasa.github.io/cumulus/docs/features/dead_letter_archive)).   The endpoint will invoke an async operation that will attempt to process all of the objects in the specified location.
```

🐜

Author: 12cd2ee -- git UI isn't committing this (conflicts?)

> The endpoint by default will process all records from the default storage location on the S3 system bucket `<stackName>/dead-letter-archive/sqs/`, however it is likely useful to process a subset of those granules by moving the records to process to another prefix on S3

Reviewer: ```suggestion
The endpoint by default will process all objects from the default storage location on the S3 system bucket under the prefix of `<stackName>/dead-letter-archive/sqs/`. However, it is likely useful to process a subset of those objects, which you can do by moving the desired subset of objects to a new path on S3 and then specifying that new path using the `path` parameter to the endpoint request.
```

🐜

Author: Disagree on s/objects/records.       While they're S3 *objects*, the thing that's being processed is a set of *records*, and from a user context the distinction matters.    We could say something like 'all records contained in the S3 objects from the default storage location', for example. 

The rest looks good.   Thoughts?

Author: try 51947ce  on for size, totally open to something else.

> The endpoint by default will process all all records contained in the S3 objects from the default storage location on the S3 system bucket under the prefix of `<stackName>/dead-letter-archive/sqs/`. However, it is likely useful to process a subset of those objects, which you can do by moving the desired subset of objects to a new path on S3 and then specifying that new path using the `path` parameter to the endpoint request.

Reviewer: ```suggestion
The endpoint by default will process all records contained in the S3 objects from the default storage location on the S3 system bucket under the prefix of `<stackName>/dead-letter-archive/sqs/`. However, it is likely useful to process a subset of those objects, which you can do by moving the desired subset of objects to a new path on S3 and then specifying that new path using the `path` parameter to the endpoint request.
```

The query uses the following optional parameters:

| parameter | description |
| --- | --- |
| bucket | The bucket to read records from. Default the Core system bucket|
| path | The S3 prefix (path) to read DLQ records from.  Defaults to `<stackName>/dead-letter-archive/sqs/`|

```endpoint

> POST /deadLetterARchive/recoverCumulusMessages

Reviewer 2: Can you make the R in archive lowercase.

Author: 👍🏻  thanks
